roles/servicetelemetry/tasks/component_clouds.yml

```diff
@@ -19,20 +19,51 @@
     - this_cloud.metrics.collectors is iterable
 
 - name: Events Smart Gateway deployment
+  when: servicetelemetry_vars.backends.events.elasticsearch.enabled
   block:
     - name: Lookup ElasticSearch BasicAuth
       k8s_info:
         api_version: v1
         kind: Secret
         namespace: '{{ ansible_operator_meta.namespace }}'
-        name: 'elasticsearch-es-elastic-user'
+        name: '{{ servicetelemetry_vars.backends.events.elasticsearch.extUserSecretName }}'
       register: elasticsearch_es_elastic_user
 
-    - name: Filter out ElasticSearch password for BasicAuth
+    - name: Set ElasticSearch user for BasicAuth
       set_fact:
-        elastic_pass: "{{ elasticsearch_es_elastic_user | json_query('resources[0].data.elastic') | b64decode }}"
+        elastic_user: "{{ elasticsearch_es_elastic_user.resources[0].data | dict2items | map(attribute='key') | list | first }}"
+
+    - name: Set ElasticSearch password for BasicAuth
+      set_fact:
+        elastic_pass: "{{ elasticsearch_es_elastic_user.resources[0].data[elastic_user] | b64decode }}"
       no_log: true
 
+    - name: Get the ElasticSearch TLS materials secret
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        namespace: '{{ ansible_operator_meta.namespace }}'
+        name: '{{ servicetelemetry_vars.backends.events.elasticsearch.extTlsSecretName }}'
+      register: es_certs
+
+    - when: es_certs[0].data[user.key] is not defined or es_certs[0].data[user.crt] is not defined
+      block:
+        - name: Load dummy certs
+          include_vars:
+            file: dummy_user_certs.yml
+
+        - name: Augment the secret with dummy TLS cert/key if no TLS user auth material provided
+          k8s:
+            definition:
+              api_version: v1
```

Comment:
When you use `definition` you use the actual k8s manifest format, so in this case it's `apiVersion` and not the Ansible parameter `api_version`.